model/model.js

```diff
@@ -14,9 +23,30 @@ exports.selectAllEndpoints = () => {
 
 exports.selectAllTopics = async () => {
   try {
-    return db.query(`SELECT * FROM topics;`);
+    return db.query(`SELECT * FROM topics;`)
+    .then((topics) => {
+      return topics.rows
+    })
   } 
   catch(err) {
     return err
   }
+}
+
+exports.selectArticleById = async (articleId) => {
+  articleId = Number(articleId)
+  const articleIdList = await getListOfArticleIds()
+  const regex = new RegExp(/[^\d]/g)
+  if (regex.test(articleId)) {
```

Comment:
Interesting solution, but it would be more efficient to query the database and then make use of PSQL errors